### ISSUE TITLE
The registered client only supports 'token_endpoint_client_auth_method' method 'client_secret_basic'

### ISSUE THREAD

**Issue by n-connect** [authelia/authelia#6860]
### Version

v4.38.0

### Deployment Method

Bare-metal

### Reverse Proxy

NGINX

### Reverse Proxy Version

1.24.0

### Description

First of all thank you for this wonderful SSO solution!

Have a working Authelia since version v4.37.5, LDAP backend. Configuration is not with secrets, but direct generated pws/tokens/keys in the configuration file, (env variables in FreeBSD rc.d-service file just got into the main).

After using it for a while, looking for RBAC, got the v4.38.0-beta3 (authelia version untagged-v4.37.5 (v4.38.0-beta3, e26aa48)) with a working RBAC authorization_policy config, looking for v4.38.0 get released.

The client web-application has been set before, then with RBAC, finally moved to v.4.38.0 today:
- the deny route drops as intended, but of course that not expands the same functions as the green route.
- the success route of authorization policy fails,

Using v4.38.0:
If I set up a user with a deny group membership only, I got a proper answer, e.g.: nothing happens, after Authelia login+2FA it drops back to the client's login screen.
However if the user has the group membership for allowing the login too, it fails with Internal server error. Reloading the page drops back the web-application/client's main page. Debug log in the log section.

Replacing back the authelia binary from 'authelia version v4.38.0' to 'authelia version untagged-v4.37.5 (v4.38.0-beta3, e26aa48)' the allowed group based login process works again, along with the deny route too. Debug log in the log section

### Reproduction

Install Authelia v4.38.0, LDAP backend (Bare-metal, FreeBSD amd64). Reverse proxy, LDAP, transparent in the issue. Webapplication as OAUTH2.0 client is HedgeDoc v1.99 - but based on error message, it can be generic with plain [generated] passwords in Authelia config. 

### Expectations

Let both ways work in v4.38.0 (or upper) as in beta6:
- not just the deny direction (appdeny_au group membership),
- but the allow route too (appallow_au group membership)

### Configuration (Authelia)

```yaml
authelia.yml (snippet):
`    authorization_policies:
      appname_policy:
        default_policy: 'two_factor'
        rules:
          - policy: 'deny'
            subject: 'group:appdeny_au'
          - policy: 'two_factor'
            subject: 'group:appallow_au'
    clients:
      - id: appname
        description: AppName SSO
        secret: '$plaintext$asdassdassdasdASDASDASDASDDASD'
        public: false
        authorization_policy: appname_policy
        audience: []
        scopes:
          - openid
          - email
          - profile
        redirect_uris:
          - https://appname.example.com/auth/oauth2/callback
        grant_types:
          - refresh_token
          - authorization_code
        response_types:
          - code
        response_modes:
          - form_post
          - query
          - fragment
        userinfo_signing_algorithm: none`
```

### Build Information

```shell
authelia_v4.38.0 build-info
Last Tag: v4.38.0
State: tagged clean
Branch: v4.38.0
Commit: a53395fd7e4f40f0ac26e29a0b722c52c3015a06
Build Number: 27593
Build OS: freebsd
Build Arch: amd64
Build Compiler: gc
Build Date: Thu, 14 Mar 2024 22:12:25 +1100
Extra:

Go:
    Version: go1.22.1
    Module Path: github.com/authelia/authelia/v4
    Executable Path: github.com/authelia/authelia/v4/cmd/authelia
    Settings:
        -buildmode: pie
        -compiler: gc
        -trimpath: true
        DefaultGODEBUG: httplaxcontentlength=1,httpmuxgo121=1,tls10server=1,tlsrsakex=1,tlsunsafeekm=1
        CGO_ENABLED: 1
        GOARCH: amd64
        GOOS: freebsd
        GOAMD64: v1
        vcs: git
        vcs.revision: a53395fd7e4f40f0ac26e29a0b722c52c3015a06
        vcs.time: 2024-03-14T10:49:39Z
        vcs.modified: true
    Dependencies:
        authelia.com/provider/oauth2@v0.0.0-20240313001612-43d016545b81 (h1:BmwFLBmC6kbZC4++6Vzfq19r/xkI7hJ6jiKfmDbSPzc=)
        filippo.io/edwards25519@v1.1.0 (h1:FNf4tywRC1HmFuKW5xopWpigGjJKiJSV0Cqo0cJWDaA=)
        github.com/Azure/go-ntlmssp@v0.0.0-20221128193559-754e69321358 (h1:mFRzDkZVAjdal+s7s0MwaRv9igoPqLRdzOLzw/8Xvq8=)
        github.com/Gurpartap/logrus-stack@v0.0.0-20170710170904-89c00d8a28f4 (h1:vdT7QwBhJJEVNFMBNhRSFDRCB6O16T28VhvqRgqFyn8=)
        github.com/andybalholm/brotli@v1.1.0 (h1:eLKJA0d02Lf0mVpIDgYnqXcUn0GqVmEFny3VuID1U3M=)
        github.com/asaskevich/govalidator@v0.0.0-20230301143203-a9d515a09cc2 (h1:DklsrG3dyBCFEj5IhUbnKptjxatkF07cF2ak3yi77so=)
        github.com/authelia/jsonschema@v0.1.7 (h1:RbtTeTG7GiWIrx2A+3O+b33jr/mLlSmqGYyk1w5gLNA=)
        github.com/authelia/otp@v1.0.0 (h1:X6YeBMb16CkW8fFpLBQc0ams+Ed0zw1R/5pfih/1vLU=)
        github.com/beorn7/perks@v1.0.1 (h1:VlbKKnNfV8bJzeqoa4cOKqO6bYr3WgKZxO8Z16+hsOM=)
        github.com/boombuler/barcode@v1.0.1 (h1:NDBbPmhS+EqABEs5Kg3n/5ZNjy73Pz7SIV+KCeqyXcs=)
        github.com/cespare/xxhash/v2@v2.2.0 (h1:DC2CZ1Ep5Y4k3ZQ899DldepgrayRUGE6BBZ/cd9Cj44=)
        github.com/davecgh/go-spew@v1.1.1 (h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=)
        github.com/dgraph-io/ristretto@v0.1.1 (h1:6CWw5tJNgpegArSHpNHJKldNeq03FQCwYvfMVWajOK8=)
        github.com/dgryski/go-rendezvous@v0.0.0-20200823014737-9f7001d12a5f (h1:lO4WD4F/rVNCu3HqELle0jiPLLBs70cWOduZpkS1E78=)
        github.com/dlclark/regexp2@v1.4.0 (h1:F1rxgk7p4uKjwIQxBs9oAXe5CqrXlCduYEJvrF4u93E=)
        github.com/duosecurity/duo_api_golang@v0.0.0-20240205144049-bb361ad4ae1c (h1:xFrCg835Y/ig7iWQqyVmGFG5cd1OztnlN3rF64ltEpY=)
        github.com/dustin/go-humanize@v1.0.1 (h1:GzkhY7T5VNhEkwH0PVJgjz+fX1rhBrR7pRT3mDkpeCY=)
        github.com/facebookgo/stack@v0.0.0-20160209184415-751773369052 (h1:JWuenKqqX8nojtoVVWjGfOF9635RETekkoH6Cc9SX0A=)
        github.com/fasthttp/router@v1.5.0 (h1:3Qbbo27HAPzwbpRzgiV5V9+2faPkPt3eNuRaDV6LYDA=)
        github.com/fasthttp/session/v2@v2.5.4 (h1:SeblRaKHYQoVBjJIF1KlZD0F8QX1poA80h/KaLhNo8I=)
        github.com/fsnotify/fsnotify@v1.7.0 (h1:8JEhPFa5W2WU7YfeZzPNqzMP6Lwt7L2715Ggo0nosvA=)
        github.com/fxamacker/cbor/v2@v2.6.0 (h1:sU6J2usfADwWlYDAFhZBQ6TnLFBHxgesMrQfQgk1tWA=)
        github.com/go-asn1-ber/asn1-ber@v1.5.5 (h1:MNHlNMBDgEKD4TcKr36vQN68BA00aDfjIt3/bD50WnA=)
        github.com/go-crypt/crypt@v0.2.19 (h1:9VFKbVCuWH4cQDbjUA6fGiaHx+w0CXI19rHQGTZqESE=)
        github.com/go-crypt/x@v0.2.13 (h1:YUgKO62hIcPz11ViwHZx89g/OJhOis9+kK13ZunWpS0=)
        github.com/go-jose/go-jose/v4@v4.0.1 (h1:QVEPDE3OluqXBQZDcnNvQrInro2h0e4eqNbnZSWqS6U=)
        github.com/go-ldap/ldap/v3@v3.4.6 (h1:ert95MdbiG7aWo/oPYp9btL3KJlMPKnP58r09rI8T+A=)
        github.com/go-sql-driver/mysql@v1.8.0 (h1:UtktXaU2Nb64z/pLiGIxY4431SJ4/dR5cjMmlVHgnT4=)
        github.com/go-viper/mapstructure/v2@v2.0.0-alpha.1 (h1:TQcrn6Wq+sKGkpyPvppOz99zsMBaUOKXq6HSv655U1c=)
        github.com/go-webauthn/webauthn@v0.10.2 (h1:OG7B+DyuTytrEPFmTX503K77fqs3HDK/0Iv+z8UYbq4=)
        github.com/go-webauthn/x@v0.1.9 (h1:v1oeLmoaa+gPOaZqUdDentu6Rl7HkSSsmOT6gxEQHhE=)
        github.com/golang-jwt/jwt/v5@v5.2.1 (h1:OuVbFODueb089Lh128TAcimifWaLhJwVflnrgM17wHk=)
        github.com/golang/glog@v1.2.0 (h1:uCdmnmatrKCgMBlM4rMuJZWOkPDqdbZPnrMXDY4gI68=)
        github.com/golang/protobuf@v1.5.3 (h1:KhyjKVUg7Usr/dYsdSqoFveMYd5ko72D+zANwlG1mmg=)
        github.com/google/go-tpm@v0.9.0 (h1:sQF6YqWMi+SCXpsmS3fd21oPy/vSddwZry4JnmltHVk=)
        github.com/google/uuid@v1.6.0 (h1:NIvaJDMOsjHA8n1jAhLSgzrAzy1Hgr+hNrb57e+94F0=)
        github.com/hashicorp/go-cleanhttp@v0.5.2 (h1:035FKYIWjmULyFRBKPs8TBQoi0x6d9G4xc9neXJWAZQ=)
        github.com/hashicorp/go-retryablehttp@v0.7.5 (h1:bJj+Pj19UZMIweq/iie+1u5YCdGrnxCT9yvm0e+Nd5M=)
        github.com/iancoleman/orderedmap@v0.3.0 (h1:5cbR2grmZR/DiVt+VJopEhtVs9YGInGIxAoMJn+Ichc=)
        github.com/jackc/pgpassfile@v1.0.0 (h1:/6Hmqy13Ss2zCq62VdNG8tM1wchn8zjSGOBJ6icpsIM=)
        github.com/jackc/pgservicefile@v0.0.0-20221227161230-091c0ba34f0a (h1:bbPeKD0xmW/Y25WS6cokEszi5g+S0QxI/d45PkRi7Nk=)
        github.com/jackc/pgx/v5@v5.5.5 (h1:amBjrZVmksIdNjxGW/IiIMzxMKZFelXbUoPNb+8sjQw=)
        github.com/jackc/puddle/v2@v2.2.1 (h1:RhxXJtFG022u4ibrCSMSiu5aOq1i77R3OHKNJj77OAk=)
        github.com/jmoiron/sqlx@v1.3.5 (h1:vFFPA71p1o5gAeqtEAwLU4dnX2napprKtHr7PYIcN3g=)
        github.com/klauspost/compress@v1.17.6 (h1:60eq2E/jlfwQXtvZEeBUYADs+BwKBWURIY+Gj2eRGjI=)
        github.com/knadh/koanf/maps@v0.1.1 (h1:G5TjmUh2D7G2YWf5SQQqSiHRJEjaicvU0KpypqB3NIs=)
        github.com/knadh/koanf/parsers/yaml@v0.1.0 (h1:ZZ8/iGfRLvKSaMEECEBPM1HQslrZADk8fP1XFUxVI5w=)
        github.com/knadh/koanf/providers/confmap@v0.1.0 (h1:gOkxhHkemwG4LezxxN8DMOFopOPghxRVp7JbIvdvqzU=)
        github.com/knadh/koanf/providers/env@v0.1.0 (h1:LqKteXqfOWyx5Ab9VfGHmjY9BvRXi+clwyZozgVRiKg=)
        github.com/knadh/koanf/providers/posflag@v0.1.0 (h1:mKJlLrKPcAP7Ootf4pBZWJ6J+4wHYujwipe7Ie3qW6U=)
        github.com/knadh/koanf/v2@v2.1.0 (h1:eh4QmHHBuU8BybfIJ8mB8K8gsGCD/AUQTdwGq/GzId8=)
        github.com/mattn/go-sqlite3@v1.14.22 (h1:2gZY6PC6kBnID23Tichd1K+Z0oS6nE/XwU+Vz/5o4kU=)
        github.com/mitchellh/copystructure@v1.2.0 (h1:vpKXTN4ewci03Vljg/q9QvCGUDttBOGBIa15WveJJGw=)
        github.com/mitchellh/mapstructure@v1.5.0 (h1:jeMsZIYE/09sWLaz43PL7Gy6RuMjD2eJVyuac5Z2hdY=)
        github.com/mitchellh/reflectwalk@v1.0.2 (h1:G2LzWKi524PWgd3mLHV8Y5k7s6XUvT0Gef6zxSIeXaQ=)
        github.com/mohae/deepcopy@v0.0.0-20170929034955-c48cc78d4826 (h1:RWengNIwukTxcDr9M+97sNutRR1RKhG96O6jWumTTnw=)
        github.com/ory/herodot@v0.10.3-0.20230807143059-27cd6936499b (h1:AEUyF55UrqTuhJh72I9azACdJrRrDBBjK/XWgVxuQvY=)
        github.com/ory/x@v0.0.616 (h1:iaojp7MvFW1cdirSZFK/XeuJvyhUEVXQdY61bmIOkzk=)
        github.com/philhofer/fwd@v1.1.2 (h1:bnDivRJ1EWPjUIRXV5KfORO897HTbpFAQddBdE8t7Gw=)
        github.com/pkg/errors@v0.9.1 (h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=)
        github.com/pmezard/go-difflib@v1.0.0 (h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=)
        github.com/prometheus/client_golang@v1.19.0 (h1:ygXvpU1AoN1MhdzckN+PyD9QJOSD4x7kmXYlnfbA6JU=)
        github.com/prometheus/client_model@v0.5.0 (h1:VQw1hfvPvk3Uv6Qf29VrPF32JB6rtbgI6cYPYQjL0Qw=)
        github.com/prometheus/common@v0.48.0 (h1:QO8U2CdOzSn1BBsmXJXduaaW+dY/5QLjfB8svtSzKKE=)
        github.com/prometheus/procfs@v0.12.0 (h1:jluTpSng7V9hY0O2R9DzzJHYb2xULk9VTR1V1R/k6Bo=)
        github.com/redis/go-redis/v9@v9.5.1 (h1:H1X4D3yHPaYrkL5X06Wh6xNVM/pX0Ft4RV0vMGvLBh8=)
        github.com/savsgio/gotils@v0.0.0-20240303185622-093b76447511 (h1:KanIMPX0QdEdB4R3CiimCAbxFrhB3j7h0/OvpYGVQa8=)
        github.com/sirupsen/logrus@v1.9.3 (h1:dueUQJ1C2q9oE3F7wvmSGAaVtTmUizReu6fjN8uqzbQ=)
        github.com/spf13/cobra@v1.8.0 (h1:7aJaZx1B85qltLMc546zn58BxxfZdR/W22ej9CFoEf0=)
        github.com/spf13/pflag@v1.0.5 (h1:iy+VFUOCP1a+8yFto/drg2CJ5u0yRoB7fZw3DKv/JXA=)
        github.com/stretchr/testify@v1.9.0 (h1:HtqpIVDClZ4nwg75+f6Lvsy/wHu+3BoSGCbBAcpTsTg=)
        github.com/tinylib/msgp@v1.1.9 (h1:SHf3yoO2sGA0veCJeCBYLHuttAVFHGm2RHgNodW7wQU=)
        github.com/trustelem/zxcvbn@v1.0.1 (h1:mp4JFtzdDYGj9WYSD3KQSkwwUumWNFzXaAjckaTYpsc=)
        github.com/valyala/bytebufferpool@v1.0.0 (h1:GqA5TC/0021Y/b9FG4Oi9Mr3q7XYx6KllzawFIhcdPw=)
        github.com/valyala/fasthttp@v1.52.0 (h1:wqBQpxH71XW0e2g+Og4dzQM8pk34aFYlA1Ga8db7gU0=)
        github.com/wneessen/go-mail@v0.4.1 (h1:m2rSg/sc8FZQCdtrV5M8ymHYOFrC6KJAQAIcgrXvqoo=)
        github.com/x448/float16@v0.8.4 (h1:qLwI1I70+NjRFUR3zs1JPUCgaCXSh3SW62uAKT1mSBM=)
        golang.org/x/crypto@v0.21.0 (h1:X31++rzVUdKhX5sWmSOFZxx8UW/ldWx55cbf08iNAMA=)
        golang.org/x/net@v0.22.0 (h1:9sGLhx7iRIHEiX0oAJ3MRZMUCElJgy7Br1nO+AMN3Tc=)
        golang.org/x/oauth2@v0.18.0 (h1:09qnuIAgzdx1XplqJvW6CQqMCtGZykZWcXzPMPUusvI=)
        golang.org/x/sync@v0.6.0 (h1:5BMeUDZ7vkXGfEr1x9B4bRcTH4lpkTkpdh0T/J+qjbQ=)
        golang.org/x/sys@v0.18.0 (h1:DBdB3niSjOA/O0blCZBqDefyWNYveAYMNF1Wum0DYQ4=)
        golang.org/x/term@v0.18.0 (h1:FcHjZXDMxI8mM3nwhX9HlKop4C0YQvCVCdwYl2wOtE8=)
        golang.org/x/text@v0.14.0 (h1:ScX5w1eTa3QqT8oi6+ziP7dTV1S2+ALU0bI+0zXKWiQ=)
        google.golang.org/genproto/googleapis/rpc@v0.0.0-20231106174013-bbf56f31fb17 (h1:Jyp0Hsi0bmHXG6k9eATXoYtjd6e2UzZ1SCn/wIupY14=)
        google.golang.org/grpc@v1.59.0 (h1:Z5Iec2pjwb+LEOqzpB2MR12/eKFhDPhuqW91O+4bwUk=)
        google.golang.org/protobuf@v1.33.0 (h1:uNO2rsAINq/JlFpSdYEKIZ0uKD/R9cpdv0T+yoGwGmI=)
        gopkg.in/yaml.v3@v3.0.1 (h1:fxVm/GzAzEWqLHuvctI91KS9hhNmmWOoWu0XTYJS7CA=)
```

### Logs (Authelia)

```shell
Working deny group membership with v4.38.0:
`time="2024-03-14T15:36:45+01:00" level=error msg="Authorization Request with id '36f7f481-633a-4b5a-8ac0-52eeefdcf394' 
on client with id 'appname' using policy 'appname_policy' could not be processed: the user 'username' is not authorized to use this client" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx `

Non-working v4.38.0 debug log for allowing membership:
`time="2024-03-14T15:47:08+01:00" level=debug msg="Authorization Request with id '0799233d-1ddb-468e-8de3-6ff887ea97ff' on client with id 'appname' is being processed" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:08+01:00" level=debug msg="Authorization Request with id '0799233d-1ddb-468e-8de3-6ff887ea97ff' on client with id 'appname' using consent mode 'explicit' proceeding to generate a new consent session" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:08+01:00" level=debug msg="Authorization Request with id '0799233d-1ddb-468e-8de3-6ff887ea97ff' on client with id 'appname' using consent mode 'explicit' authentication level 'one_factor' is insufficient for client level 'two_factor'" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:08+01:00" level=debug msg="Authorization Request with id '0799233d-1ddb-468e-8de3-6ff887ea97ff' on client with id 'appname' using consent mode 'explicit' is being redirected to 'https://auth.example.com/?workflow=openid_connect&workflow_id=2488e35f-4cb9-488f-96d8-aeb7aeb559e8'" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:20+01:00" level=debug msg="Mark TOTP authentication attempt made by user 'attis'" method=POST path=/api/secondfactor/totp remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:20+01:00" level=debug msg="Successful TOTP authentication attempt made by user 'attis'" method=POST path=/api/secondfactor/totp remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:21+01:00" level=debug msg="Authorization Request with id '203f8d72-c9d5-4219-82bc-a2490ffcb9ee' on client with id 'appname' is being processed" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:21+01:00" level=debug msg="Authorization Request with id '203f8d72-c9d5-4219-82bc-a2490ffcb9ee' on client with id 'appname' using consent mode 'explicit' authentication level 'two_factor' is sufficient for client level 'two_factor'" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:21+01:00" level=debug msg="Authorization Request with id '203f8d72-c9d5-4219-82bc-a2490ffcb9ee' on client with id 'appname' using consent mode 'explicit' is being redirected to 'https://auth.example.com/consent?id=2488e35f-4cb9-488f-96d8-aeb7aeb559e8'" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:24+01:00" level=debug msg="Authorization Request with id 'ead55c3d-f4aa-467f-aad5-5b52048666c7' on client with id 'appname' is being processed" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:24+01:00" level=debug msg="Authorization Request with id 'ead55c3d-f4aa-467f-aad5-5b52048666c7' on client with id 'appname' was successfully processed, proceeding to build Authorization Response" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T15:47:24+01:00" level=error msg="Access Request failed with error: Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). The request was determined to be using 'token_endpoint_client_auth_method' method 'client_secret_post', however the OAuth 2.0 client does not support this method. The registered client with id 'appname' only supports 'token_endpoint_client_auth_method' method 'client_secret_basic'." method=POST path=/api/oidc/token remote_ip=130.255.77.37

stack="github.com/authelia/authelia/v4/internal/handlers/handler_oidc_token.go:25 
OpenIDConnectTokenPOST\ngithub.com/authelia/authelia/v4/internal/middlewares/http_to_authelia_handler_adaptor.go:114 handleRouter.NewHTTPToAutheliaHandlerAdaptor.func25\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54 
handleRouter.(*BridgeBuilder).Build.func16.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:82 
SecurityHeadersNoStore.func1\ngithub.com/valyala/fasthttp@v1.52.0/userdata.go:5 
(*userData).Get\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:12 
SecurityHeaders.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/cors.go:216 
handleRouter.(*CORSPolicy).Middleware.func26\ngithub.com/fasthttp/router@v1.5.0/router.go:441 
(*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14 
handleRouter.LogRequest.func40\ngithub.com/authelia/authelia/v4/internal/middlewares/errors.go:38 
RecoverPanic.func1\ngithub.com/valyala/fasthttp@v1.52.0/server.go:2374 
(*Server).serveConn\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:224 
(*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.52.0/workerpool.go:196 
(*workerPool).getCh.func1\nruntime/asm_amd64.s:1695 
goexit"`

Working debug log after going back to autheliav4.38.0-beta3, e26aa48, for both deny and allow routes with same user - just playing with group memberships:
`...
time="2024-03-14T16:25:15+01:00" level=debug msg="Registering client appname with policy appname_policy (deny)"
...
time="2024-03-14T16:26:56+01:00" level=debug msg="Authorization Request with id '630be3a1-9077-41bb-98e9-031daefe81eb' on client with id 'appname' is being processed" method=GET path=/api/oidc/authorize remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:56+01:00" level=debug msg="Authorization Request with id '630be3a1-9077-41bb-98e9-031daefe81eb' on client with id 'appname' using consent mode 'explicit' proceeding to generate a new consent session" method=GET path=/api/oidc/authorize remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:56+01:00" level=debug msg="Authorization Request with id '630be3a1-9077-41bb-98e9-031daefe81eb' on client with id 'appname' using consent mode 'explicit' authentication level 'two_factor' is sufficient for client level 'two_factor'" method=GET path=/api/oidc/authorize remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:56+01:00" level=debug msg="Authorization Request with id '630be3a1-9077-41bb-98e9-031daefe81eb' on client with id 'appname' using consent mode 'explicit' is being redirected to 'https://auth.nconnect.hu/consent?id=9e66f035-21a5-44eb-84a0-414e83d0887a'" method=GET path=/api/oidc/authorize remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:58+01:00" level=debug msg="Authorization Request with id 'e16ea55e-c175-4a43-8e62-50cc2e7fb872' on client with id 'appname' is being processed" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:58+01:00" level=debug msg="Authorization Request with id 'e16ea55e-c175-4a43-8e62-50cc2e7fb872' on client with id 'appname' was successfully processed, proceeding to build Authorization Response" method=GET path=/api/oidc/authorization remote_ip=xx.xxx.xx.xxx
time="2024-03-14T16:26:58+01:00" level=debug msg="Access Request with id 'e16ea55e-c175-4a43-8e62-50cc2e7fb872' on client with id 'appname' is being processed" method=POST path=/api/oidc/token remote_ip=yyy.yyy.yy.yy
time="2024-03-14T16:26:58+01:00" level=debug msg="Access Request with id 'e16ea55e-c175-4a43-8e62-50cc2e7fb872' on client with id 'appname' has successfully been processed" method=POST path=/api/oidc/token remote_ip=yyy.yyy.yy.yy
time="2024-03-14T16:26:58+01:00" level=debug msg="UserInfo Request with id 'd60ba834-7993-4726-8fdd-2566c7e585ed' is being processed" method=GET path=/api/oidc/userinfo remote_ip=yyy.yyy.yy.yy
time="2024-03-14T16:26:58+01:00" level=debug msg="UserInfo Request with id 'd60ba834-7993-4726-8fdd-2566c7e585ed' on client with id 'appname' is being returned unsigned as per the registered client configuration" method=GET path=/api/oidc/userinfo remote_ip=yyy.yyy.yy.yy
time="2024-03-14T16:26:58+01:00" level=debug msg="UserInfo Request with id 'd60ba834-7993-4726-8fdd-2566c7e585ed' on client with id 'appname' was successfully processed" method=GET path=/api/oidc/userinfo remote_ip=yyy.yyy.yy.yy`
```

### Documentation

`identity_providers:
  oidc:
    authorization_policies:
      policy_name:
        default_policy: 'two_factor'
        rules:
          - policy: 'deny'
            subject: 'group:services'
    clients:
      - client_id: 'client_with_policy_name'
        authorization_policy: 'policy_name'`

[OIDC Provider.md](https://github.com/authelia/authelia/blob/master/docs/content/configuration/identity-providers/openid-connect/provider.md)

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by n-connect** [comment]
Nice AI answer :)
Right know it is configured with `client_secret_basic`, explained in the description. Always was like that.

I'm not sure it this was clear: without any config change, the error only exist with the fresh v.4.80.0 release. With its beta release cousin "authelia version untagged-v4.37.5 (v4.38.0-beta3, e26aa48)" it works. So right now it is running with the beta3. 
Also it seems independent of the authorization policy/RBAC config. If the policy configuration for the client changed to `two_factor`, e.g. go for the always success/green route - it always fails the same way - as its always the allowed way without RBAC.
Finally bumped into another issue, but will wait for this to make sure not to raise another one unnecessary.

Curious if anybody else has this issue. Happy to test any test/fix release, it is a simple binary replacement.

**Comment by james-d-elliott** [comment]
You'll need to configure it, the error is very clear and it's noted in the release notes as an important change.

**Comment by n-connect** [comment]
Hi @james-d-elliott 

Thanks for the heads up. I've still did found is as `token_endpoint_client_auth_method`, after removing "client_" finally seen it.

For anyone who bumps into the same problem, the direct links with the available config sample and the config item description: [token_endpoint_auth_method](https://www.authelia.com/configuration/identity-providers/openid-connect/clients/#token_endpoint_auth_method) , but it is missing the `client` tag.

I've followed the above link and I've added it to the config to -> oicd: -> clients: -> id: section for the particular client as follows:
`        authorization_policy: hdoc_policy`
`        token_endpoint_auth_method: client_secret_basic`

Than I've rechecked the release details again. I think there's still an issue here:
- in my config `token_endpoint_auth_method` or `token_endpoint_client_auth_method` was not exist previously
- if I put in `token_endpoint_client_auth_method` the service fails to start `time="2024-03-14T21:13:42+01:00" level=error msg="Configuration: configuration key not expected: identity_providers.oidc.clients[].token_endpoint_client_auth_method"`
- the release says: "If no method is defined the confidential client type will default to `client_secret_basic` as this is required by the specification." Based on my tests, it seems this is not the case.
- even after I have put `token_endpoint_auth_method: client_secret_basic` into te config explicitly, the problem persist, and the error is the same, debug log says the same. So I can't change the behaviour with it.

**Based on the above I think somehow between the beta3 and the final release of 4.38.0 the default changed to** `client_secret_post` Fix me if I'm wrong somewhere please.

Debug log:
`time="2024-03-14T21:19:55+01:00" level=error msg="Access Request failed with error: Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentica│
tion method). The request was determined to be using 'token_endpoint_client_auth_method' method 'client_secret_post', however the OAuth 2.0 client does not support this method. The registered client with id 'appname' only supports 'token_endpoint_client_auth_method' method 'client_secret_basic'.`

**Comment by james-d-elliott** [comment]
The default method is `client_secret_basic`. Change it to the method the client is trying to use. Open a discussion if you want support.

**Comment by james-d-elliott** [comment]
Sorry for being a bit irritable, was under a lot of pressure. Still welcome a discussion if you're having issues.

**Comment by james-d-elliott** [comment]
@n-connect apologies for being a bit irritable, especially when you were so nice, was under a lot of pressure. Still welcome a discussion if you're having issues.

**Comment by n-connect** [comment]
@james-d-elliott 

No problem, forget about, fixed it myself. And as this issue was closed already and writing post were restricted to contributors only too, I did not took the time to write it down - finding/testing the fix took several hours already. Beside that, I was really happy to see the RBAC capability got into official release.
I've seen You were busy too or even more:  releasing 3 versions in the past ~2 days by continuously fixing issues. Followed the new releases and I can confirm the `      token_endpoint_auth_method: 'client_secret_post'` works for all releases till v4.38.2.

The misleading error message make it to eat more time, I'm not an Oauth 2.0/OIDC coder for sure. First, downgrading was the temporary solution for me to get Authelia running, proving it to be an actual breaking change.

I think "advertising" this particular change better in documentation, with such sample config as it is in already in #6884 will cost less time for the project in the future. Just 2 days and the 2nd issue raised already under a completely different name. 

Went trough the stack part from debug log message, then 'grepped' the whole code for parts of the error message to make a PR, but until now I've did not find where these messages comes from in the code - looked in the [authelia/oauth2-client](https://github.com/authelia/oauth2-client) too. If you can point me there, I'm happy to contribute, as the error message has real contradictions with key false parts in the end. Maybe its easier to recognize for people who are not knee deep in coding it, eg. has less knowledge, and read things first, getting it as it written.

Let me show you breaking down the log message, what it says for me (and perhaps to others too): 

time="2024-03-14T21:19:55+01:00" level=error msg="Access Request failed with error: Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method).  -> true no question

`The request was determined to be using 'token_endpoint_client_auth_method' method 'client_secret_post',` -> that's true again, but only confirmed to me after the what fix worked
`however the OAuth 2.0 client does not support this method.` -> based on the actual fix for the problem is to put into the Authelia config this `token_endpoint_auth_method: client_secret_post`` it must a wrong statement. Also it is not clear what OAuth 2.0 client the error message talks about.Is it Authelia's own one or the external webapplication? Anyway the actual fix negates this statement in both cases (fixme if I'm wrong).
`The registered client with id 'appname' only supports 'token_endpoint_client_auth_method' method 'client_secret_basic'.` -> The biggest final false part. This was the first I tried with an explicit config line, wrote in my previous post. 

In short, this log message is a twisted catch of 22. If the new default is the _basic_ token which you should not specify at all in the config and my _[client_]id:_ is support only the very same _basic_, why is there an error at all. And how is my client has sent a _post_? Which is not supported by the Oauth 2.0 client anyway :D 

As this is an already closed issue, I'll put the key parts into #6884 too
